lib/internal/http2/core.js

```diff
@@ -1154,6 +1168,42 @@ class ServerHttp2Session extends Http2Session {
   get server() {
     return this[kServer];
   }
+
+  // Submits an altsvc frame to be sent to the client. `stream` is a
+  // numeric Stream ID. origin is a URL string that will be used to get
+  // the origin. alt is a string containing the altsvc details. No fancy
+  // API is provided for that.
+  altsvc(alt, originOrStream) {
+    if (this.destroyed)
+      throw new errors.Error('ERR_HTTP2_INVALID_SESSION');
+
+    let stream = 0;
+    let origin;
+
+    if (typeof originOrStream === 'string') {
+      origin = (new URL(originOrStream)).origin;
```

Comment:
I think we would be better of in accepting a `URL` object rather than a string, or an object with an `origin`  property. The typical usage is to send this down for every session, and passing an object allows a user to avoid parsing a URL for every request.

Reply:
Let's support both. If `originOrStream` is passed in as an object with an `origin` property, it will be used. This risks, of course, the value not being strictly ASCII, but we can document that requirement.